lmr

```diff
@@ -60,57 +48,50 @@ function clean_up() {
         rm -r "$TEMPDIR" &&
         echo  $' \e[1;32m>>>\e[m' Temporary directory deleted: $'\e[1;32m'"$TEMPDIR"$'\e[m' ||
         echo  $' \e[1;31m*\e[m' Failed to delete temporary directoy: $'\e[1;32m'"$TEMPDIR"$'\e[m'
+    else
+        echo  $' \e[1;33m*\e[m' Mapper output directory: $'\e[1;32m'"$TEMPDIR"$'\e[m'
     fi
-    rm "$HASHING_SCRIPT"
+    rm "$HASHING_SCRIPT" &&
+    echo  $' \e[1;32m>>>\e[m' Temporary hashing script deleted: $'\e[1;32m'"$HASHING_SCRIPT"$'\e[m' ||
+    echo  $' \e[1;31m*\e[m' Failed to delete hashing script: $'\e[1;32m'"$HASHING_SCRIPT"$'\e[m'
 } >&2
-trap 'clean_up; exit' SIGHUP SIGINT SIGTERM 
+trap 'clean_up' EXIT
+trap 'exit' SIGHUP SIGINT SIGTERM
 
 cat <<EOF > "${HASHING_SCRIPT}"
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-from __future__ import print_function, unicode_literals
-import sys, gzip, re, os, fileinput, io
-def tokens(str1): return re.findall('[a-z]+', str1.lower())
-N_REDUCER, MAPPER_ID, BASE_DIR  = int(sys.argv[1]), int(sys.argv[2]), sys.argv[3]
-
-# print(MAPPER_ID, ' ', file = sys.stderr, end = '')
-# print(MAPPER_ID, end = ' ')
+import sys, os, fileinput
 
-import base64
-def outfile(seg_id):
-    segdir = '{}/reducer-{:02}'.format(BASE_DIR, seg_id)
-    try: os.makedirs(segdir)
-    except OSError : pass
-    # return io.open('{}/{:02}'.format( segdir, base64.urlsafe_b64encode() ), 'at', buffering=1, encoding=None)
-    # return io.open('{}/mapper-{:02}'.format( segdir, MAPPER_ID ), 'at', buffering=1, encoding=None)
-    return io.open('{}/mapper-{:02}'.format( segdir, MAPPER_ID ), 'wt', encoding = 'utf-8')
+N_REDUCER, MAPPER_ID, BASE_DIR  = int(sys.argv[1]), int(sys.argv[2]), sys.argv[3]
 
-seg_file = [ outfile(seg_id) for seg_id in range(N_REDUCER) ]
+seg_file = [open(os.path.join(BASE_DIR, f"reducer-{seg_id:02}", f"mapper-{MAPPER_ID:02}"), 'w')
+            for seg_id in range(N_REDUCER)]
 
 for line in fileinput.input(sys.argv[4:]):
-    key, _,value = line[:-1].partition('\t')
-    print( '{}\t{}'.format(key, value), file = seg_file[hash(key) % N_REDUCER])
-fileinput.close()
-for seg_id in range(N_REDUCER): seg_file[seg_id].close()
+    key, _, value = line.rstrip().partition('\t')
+    print(key, value, sep='\t', file=seg_file[hash(key) % N_REDUCER])
```

Comment:
you are still using `hash()`?
I thought this PR is for replacing the hash function with md5 or other stable hash function

Reply:
Hi, I turned back to use built-in `hash()` for performance reason. #b81c071

> Python's hash function is designed for speed, and maps into a 64-bit space.
https://stackoverflow.com/questions/7646520/

And by setting `PYTHONHASHSEED` to a fixed number, I can get consistent result across processes.
https://docs.python.org/3/using/cmdline.html#envvar-PYTHONHASHSEED

I simply use 0 to disable hash randomization.
https://github.com/dspp779/local-mapreduce/commit/b81c07141f0cda90dc7bf0883313be1e6f5e8882#diff-6f9247eb06cc4a7c1c762555945b1aafR97